Add Alternative block placement Protocol

Bacteriawa opened the pull request:
#78

Bacteriawa committed:
Add Alternative block placement Protocol

Suisuroru committed:
[ci skip] update datapack command save

Suisuroru commented:
Enum的不会被自动转换

Bacteriawa commented:
我单独写一个Enum的类吧

Suisuroru commented:
> 我单独写一个Enum的类吧

不用了，直接加个方法，然后改成String，然后用方法获取值

Suisuroru committed:
Merge remote-tracking branch 'origin/ver/1.21.8' into fork/Bacteriawa/ver/1.21.8

Suisuroru committed:
fix: fix config error caused by config system

- TODO: fix config system read enum bug in luminol

Suisuroru commented:
<img width="300" height="55" alt="image" src="https://github.com/user-attachments/assets/9d6abc78-95bb-4616-86de-464c454578a0" />

Suisuroru commented:
@TT-Candy 有空测一下（

mc2514 commented:
cce也会一起合并嘛

Suisuroru commented:
666，我勒个无法放置啊






------------------&nbsp;Original&nbsp;------------------
From:                                                                                                                        "LuminolMC/Lophine"                                                                                    ***@***.***&gt;;
Date:&nbsp;Sat, Nov 8, 2025 09:01 PM
***@***.***&gt;;
***@***.******@***.***&gt;;
Subject:&nbsp;Re: [LuminolMC/Lophine] Add Alternative block placement Protocol (PR #81)



TT-Candy left a comment (LuminolMC/Lophine#81)
 
非常好的协议 使我无法放置任何带有方向性的方块
 以及QQ喊我比较快
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***&gt;

Suisuroru reviewed (changes requested):
实测获取到的命中点与在Leaves不一致，需要排查原因（

Suisuroru committed:
remove Leaves logger & use mojang logger

Suisuroru committed:
Merge branch 'ver/1.21.8' into ver/1.21.8

Suisuroru committed:
Merge remote-tracking branch 'origin/ver/1.21.8' into fork/Bacteriawa/ver/1.21.8

Bacteriawa commented:
我没招了，看不出哪里出问题（）

Bacteriawa commented:
> @Bacteriawa ,你能看出来那个获取数据来源吗，这个数据和Leaves上的表现不同（
> BlockHitResult hitResult = packet.getHitResult();
> 
> net/minecraft/server/network/ServerGamePacketListenerImpl.java Line 2079

有什么不同处嘛

Suisuroru commented:
> > @Bacteriawa ,你能看出来那个获取数据来源吗，这个数据和Leaves上的表现不同（
> > BlockHitResult hitResult = packet.getHitResult();
> > net/minecraft/server/network/ServerGamePacketListenerImpl.java Line 2079
> 
> 有什么不同处嘛

没看出来啊（

你说有没有可能，这个数据与实际放置点更近了，然后后面都要用和实际点距离，太近了会判为正常放置不走轻松放置（

Suisuroru commented:
要不重写检测吧，直接用射线检测来代替（？）

Suisuroru committed:
feat: use enum instead string

Because of enum supported in this version of Luminol, so use enum directly to save AlternativePlaceType.

Bacteriawa commented:
坏消息，重写了也是一样的

Suisuroru commented:
> @Suisuroru 坏消息，重写了也是一样的

6，那爆炸了（

Suisuroru committed:
update code from leaves

Suisuroru committed:
Merge branch 'ver/1.21.8' into fork/Bacteriawa/ver/1.21.8

Suisuroru reviewed (approved):
本地测试通过（debug环境）

Suisuroru committed:
Merge branch 'ver/1.21.8' into ver/1.21.8

TT-Candy commented:
莫得问题
